spack.yaml fails to build Fails to Build due to g4vg/VecGeom CMake ALIAS Target Conflict

Hello Celeritas Team,
I am trying to build Celeritas in order to accelerate Geant4 simulations. I have been following the recommended installation path using Spack (V1.1.0.dev0), (scripts/spack.yaml) but have run into a series of build-stopping issues.

When Spack attempts to build the g4vg package, the build fails during the CMake configuration step with the following error:

-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  ```
The CXX compiler identification is GNU 12.3.0
 Detecting CXX compiler ABI info
 Detecting CXX compiler ABI info - done
 Check for working CXX compiler: /usr/bin/c++ - skipped
 Detecting CXX compile features
 Detecting CXX compile features - done
 Found Geant4: /opt/spack/var/spack/environments/celeritas/.spack-env/view/lib/cmake/Geant4/Geant4Config.cmake (found version "11.3.2")
 Looking for a CUDA compiler
 Looking for a CUDA compiler - /opt/spack/var/spack/environments/celeritas/.spack-env/view/bin/nvcc
 The CUDA compiler identification is NVIDIA 12.8.93 with host compiler GNU 12.3.0
 Detecting CUDA compiler ABI info
 Detecting CUDA compiler ABI info - done
 Check for working CUDA compiler: /opt/spack/var/spack/environments/celeritas/.spack-env/view/bin/nvcc - skipped
 Detecting CUDA compile features
 Detecting CUDA compile features - done
 Found Threads: TRUE
 Found VecGeom: /opt/spack/var/spack/environments/celeritas/.spack-env/view/lib/cmake/VecGeom/VecGeomConfig.cmake (found suitable version "1.2.11", minimum required is "1.2.8")
 CMake Error at cmake/FindVecGeom.cmake:50 (target_compile_options):
   target_compile_options can not be used on an ALIAS target.
 Call Stack (most recent call first):
   CMakeLists.txt:75 (find_package)

 CMake Error at cmake/FindVecGeom.cmake:53 (target_link_options):
   target_link_options can not be used on an ALIAS target.
 Call Stack (most recent call first):
   CMakeLists.txt:75 (find_package)

 CMake Error at cmake/FindVecGeom.cmake:50 (target_compile_options):
   target_compile_options can not be used on an ALIAS target.
 Call Stack (most recent call first):
   CMakeLists.txt:75 (find_package)

 CMake Error at cmake/FindVecGeom.cmake:53 (target_link_options):
   target_link_options can not be used on an ALIAS target.
 Call Stack (most recent call first):
   CMakeLists.txt:75 (find_package)

Configuring incomplete, errors occurred!
```

--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Attempts to Work Around
---
I have tried several workarounds, all of which have been blocked:
Other vorsions of g4vg: Using  versions (1.0.2, 1.0.3 or 1.0.4) results in the same ALIAS target error.
Downgrading VecGeom: Attempting to downgrade VecGeom in the spack.yaml to an older version (e.g., 1.2.8) got ignored by spack.

Note on CUDA/Compiler Issues
I also encountered a glibc/CUDA header incompatibility (noexcept specification error) when using the system's default GCC 14.

```
/usr/include/x86_64-linux-gnu/bits/mathcalls.h(81): error: exception
specification is incompatible with that of previous function "sinpi"
(declared at line 2556 of /home/colle/spack/opt/spack/linux-skylake/cuda-12.9.0-ztylz56f3kj6tio3uvbmsquxm2ixr77n/bin/../targets/x86_64-linux/include/crt/mathfunctions.h)
extern double sinpi (double __x) noexcept (true); extern double __sinpi (double _x) noexcept (true);
```

The workaround i found to be successful for this was to build a Docker container with a clean environment and force Spack to use gcc@12, which is known to be compatible and have spack build its own cuda (V12.8.93 cuda_arch=75). 

I have attached the Dockerfile used to create the build environment and the modified spack.yaml that triggers the error. They represent an  environment that successfully builds all dependencies except g4vg.

[Dockerfile_toolchain.txt](https://github.com/user-attachments/files/21983802/Dockerfile_toolchain.txt)

[spack.yaml](https://github.com/user-attachments/files/21983838/spack.yaml)

Steps to reproduce
-
- build docker image using the provided Dockerfile.toolchain and run it creating a virtual volume targeting the celeritas-0.6.0 folder
- create and install the spack environment using the provided spack.yaml

Thank you for your time and for this project. Any guidance on a potential fix or a recommended combination of dependency versions would be greatly appreciated.



--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
# System Details Report


## Report details
- **Date generated:**                              2025-08-26 10:22:39

## Hardware Information:
- **Hardware Model:**                              HP HP Z2 Tower G9 Workstation Desktop PC
- **Memory:**                                      32.0 GiB
- **Processor:**                                   13th Gen Intel® Core™ i9-13900 × 32
- **Graphics:**                                    NVIDIA GeForce RTX™ 2080 Ti
- **Graphics 1:**                                  Intel Corporation Raptor Lake-S GT1 [UHD Graphics 770]
- **Disk Capacity:**                               2.0 TB

## Software Information:
- **Firmware Version:**                            U50 Ver. 03.02.02
- **OS Name:**                                     Ubuntu 25.04
- **OS Build:**                                    (null)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               48
- **Windowing System:**                            Wayland
- **Kernel Version:**                              Linux 6.14.0-28-generic

-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

--- a/repos/spack_repo/builtin/packages/g4vg/package.py
+++ b/repos/spack_repo/builtin/packages/g4vg/package.py
@@ -20,6 +20,7 @@ class G4vg(CMakePackage):
 
     version("develop", branch="main", get_full_repo=True)
 
+    version("1.0.5", sha256="fa44d071aaf95de596f7b22d2062830acd7e154a3bc9d67ad5a306eff64630d2")
     version("1.0.4", sha256="ac38878aedaf9fe628f498290cf4fa6cdd21980b7b83663ce8ef3c8f3ff31cb2")
     version("1.0.3", sha256="6f0920a9ad2e04a701bec636f117d4093be1e50761f091ec507efd078b659bcd")
     version("1.0.2", sha256="daeb9263f2741c4a1073eb26f2e9fc10e89207c1ed3425da70db934069ff4160")
@@ -29,7 +30,8 @@ class G4vg(CMakePackage):
     variant("shared", default=True, description="Build shared libraries")
 
     depends_on("cxx", type="build")
-    depends_on("vecgeom@1.2.8:")
+    depends_on("vecgeom@1.2.7:", when="@1.0.5:")
+    depends_on("vecgeom@1.2.8:", when="@:1.0.4")
     depends_on("geant4")
 
     def cmake_args(self):

